Source/Core/Core/PowerPC/JitArm64/JitAsm.cpp

```diff
@@ -97,32 +97,23 @@ void JitArm64::GenerateAsm()
 
   if (assembly_dispatcher)
   {
-    if (GetBlockCache()->GetFastBlockMap())
+    if (GetBlockCache()->GetEntryPoints())
     {
       // Check if there is a block
-      ARM64Reg pc_masked = ARM64Reg::X25;
-      ARM64Reg cache_base = ARM64Reg::X24;
+      ARM64Reg pc_and_msr = ARM64Reg::X25;
+      ARM64Reg cache_base = ARM64Reg::X27;
       ARM64Reg block = ARM64Reg::X30;
-      LSL(pc_masked, DISPATCHER_PC, 1);
-      MOVP2R(cache_base, GetBlockCache()->GetFastBlockMap());
-      LDR(block, cache_base, pc_masked);
+      LDR(IndexType::Unsigned, pc_and_msr, PPC_REG, PPCSTATE_OFF(msr));
+      AND(pc_and_msr, pc_and_msr, LogicalImm(JitBaseBlockCache::JIT_CACHE_MSR_MASK, 32));
+      LDR(IndexType::Unsigned, pc_and_msr, PPC_REG, PPCSTATE_OFF(msr));
+      MOVP2R(cache_base, GetBlockCache()->GetEntryPoints());
+      // The entry points map is indexed by ((msrBits << 26) | (address >> 2)).
+      UBFIZ(pc_and_msr, pc_and_msr, 26, 6);
+      BFXIL(pc_and_msr, DISPATCHER_PC, 2, 30);
```

Comment:
Nitpick: DISPATCHER_PC should be 64-bit encoded here. (Arm64Emitter will accept it anyway though, since it actually only checks if the destination register is 64-bit.)

```suggestion
      BFXIL(pc_and_msr, EncodeRegTo64(DISPATCHER_PC), 2, 30);
```